src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2752,210 @@ def _convertTFStoOdemisMD(metadata: str) -> dict:
+
+        # TODO: try to do it similar to tifffile to extract header information too            
+        try:
+            # NOTE: 
+            # Date, Time are separated by : 
+            # everything else is separated by =
+            try:
+                k, v = line.split("=")
+            except:           
+                k, v = line.split(":", maxsplit=1) 
+            k, v = k.strip(), v.strip()
+            zmd[k] = v
+        except Exception as e:
+            continue
+
+    return zmd
+
+
+def _convertZEISStoOdemisMD(metadata: str) -> dict:
+    """Converts ZEISS metadata to odemis metadata format
+    :param metadata: metadata string
+    :return: (dict) metadata as dictionary
+    :raises: (KeyError): if the metadata cannot be parsed
+    """
+
+    # parse as dict
+    zmd = _parseZEISSMetadata(metadata)
+
+    md = {}
+
+    # convert to odemis format
+    # general
+    try:
+        datetime_str = zmd["Date"] + " " + zmd["Time"]
+        date = datetime.strptime(datetime_str, "%d %b %Y %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_ACQ_DATE} - {e}")
+    try:
+        md[model.MD_HW_NAME] = zmd["Serial No."]
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_HW_NAME} -  {e}")
+    try:
+        md[model.MD_SW_VERSION] = zmd["Version"]
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_SW_VERSION} - {e}")
+
+    try:
+        # NOTE: 
+        # There are three values in the ZEISS metadata for pixel size
+        # Pixel Size, Image Pixel Size, and FIB Pixel Size
+        # Pixel Size is the one we want as it changes based on the beam type (FIB Imaging field)
+
+        pixel_size, unit = zmd["Pixel Size"].split(" ")[-2:]
+
+        if unit == "nm":
+            pixel_size = float(pixel_size) * 1e-9
+        elif unit == "µm":
+            pixel_size = float(pixel_size) * 1e-6
+        else:
+            raise ValueError(f"Unknown unit: {unit}")
+
+        md[model.MD_PIXEL_SIZE] = (pixel_size, pixel_size)
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_PIXEL_SIZE} - {e}")
+
+    try:
+        dwell_time, unit = zmd["Dwell Time"].split(" ")[-2:]
+        if unit == "ns":
+            dwell_time = float(dwell_time) * 1e-9
+        elif unit == "µs":
+            dwell_time = float(dwell_time) * 1e-6
+        else:
+            raise ValueError(f"Unknown unit: {unit}")
+
+        md[model.MD_DWELL_TIME] = dwell_time
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_DWELL_TIME} - {e}")
+    try:
+        unit = zmd["Stage at X"].split(" ")[-1]
+        x = float(zmd["Stage at X"].split(" ")[0])
+        y = float(zmd["Stage at Y"].split(" ")[0])
+
+        # TODO: investigate using src/odemis/util/units.py
```

Comment:
Could you investigate now? I think it would be just a matter of something like that:
```
str_val, si_prefix, unit = decompose_si_prefix(str_x, "m")
x = si_scale_val(float(str_val), si_prefix)
```
This would make sure that if x and y are different units, it still reads the correct values.